app/Http/routes.php

```diff
@@ -75,8 +75,24 @@
 /*-----------------------*
  *  Profile Controller   *
  *-----------------------*/
- 
- 
+
+Route::get('profile/view/{username}'		, 'ProfileController@profileView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
+Route::get('profile/view/'			, 'ProfileController@profileViewDefault');								// Implemented
+Route::get('profile/delete/{username}'		, 'ProfileController@deleteView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');
+Route::get('profile/friends/add/{username}'	, 'ProfileController@addfriendView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');
+Route::post('profile/friends/add/{username}'	, 'ProfileController@addfriend')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');
+Route::get('profile/friends/remove/{username}'	, 'ProfileController@deletefriendView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');
+Route::post('profile/friends/remove/{username}'	, 'ProfileController@deletefriend')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');
+Route::get('profile/friends/view/{username}'	, 'ProfileController@friendView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
+Route::get('profile/friends/{username}'		, 'ProfileController@friendView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
+Route::get('profile/friends/'			, 'ProfileController@friendViewDefault');								// Implemented
+Route::get('profile/projects/{username}'	, 'ProfileController@projectView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
+Route::get('profile/projects/'			, 'ProfileController@projectViewDefault');								// Implemented
+Route::get('profile/comments/{username}'	, 'ProfileController@commentsView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
+Route::get('profile/comments/'			, 'ProfileController@commentsViewDefault');								// Implemented	
+Route::get('profile/{username}'			, 'ProfileController@profileView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
+Route::get('profile/'				, 'ProfileController@profileViewDefault');								// Implemented
+
```

Comment:
You shouldn't be limiting the passed variable here. If you want to check data and limit it do that in a controller or in a custom request.